components/new_post/index.tsx

```diff
@@ -0,0 +1,112 @@
+import {AnyAction, bindActionCreators, Dispatch} from 'redux';
+
+import {showActionsDropdownPulsatingDot} from 'selectors/actions_menu';
+import {setActionsMenuInitialisationState} from 'mattermost-redux/actions/preferences';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+
+import {
+    get,
+    isCollapsedThreadsEnabled,
+} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {getUser} from 'mattermost-redux/selectors/entities/users';
+
+import {Emoji} from '@mattermost/types/emojis';
+import {Post} from '@mattermost/types/posts';
+import {selectPost} from 'actions/views/rhs';
+
+import {markPostAsUnread, emitShortcutReactToLastPostFrom} from 'actions/post_actions';
+
+import {getShortcutReactToLastPostEmittedFrom, getOneClickReactionEmojis} from 'selectors/emojis';
+import {getIsPostBeingEditedInRHS, isEmbedVisible} from 'selectors/posts';
+import {getHighlightedPostId} from 'selectors/rhs';
+import {getIsMobileView} from 'selectors/views/browser';
+
+import {GlobalState} from 'types/store';
+
+import {isArchivedChannel} from 'utils/channel_utils';
+import {areConsecutivePostsBySameUser, shouldShowActionsMenu} from 'utils/post_utils';
+import {Preferences} from 'utils/constants';
+
+import PostComponent from './post_component';
+
+interface OwnProps {
+    post: Post;
+    previousPostId: string;
+    teamId: string;
+}
+
+function isConsecutivePost(state: GlobalState, ownProps: OwnProps) {
+    const post = ownProps.post;
+    const previousPost = ownProps.previousPostId && getPost(state, ownProps.previousPostId);
+
+    let consecutivePost = false;
+
+    if (previousPost) {
+        consecutivePost = areConsecutivePostsBySameUser(post, previousPost);
+    }
+    return consecutivePost;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    const config = getConfig(state);
+    const enableEmojiPicker = config.EnableEmojiPicker === 'true';
+    const enablePostUsernameOverride = config.EnablePostUsernameOverride === 'true';
+    const teamId = ownProps.teamId || getCurrentTeamId(state);
+    const channel = state.entities.channels.channels[ownProps.post.channel_id];
+    const shortcutReactToLastPostEmittedFrom = getShortcutReactToLastPostEmittedFrom(state);
+
+    const user = getUser(state, ownProps.post.user_id);
+    const isBot = Boolean(user && user.is_bot);
+    const highlightedPostId = getHighlightedPostId(state);
+    const showActionsMenuPulsatingDot = showActionsDropdownPulsatingDot(state);
+
+    let emojis: Emoji[] = [];
+    const oneClickReactionsEnabled = get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.ONE_CLICK_REACTIONS_ENABLED, Preferences.ONE_CLICK_REACTIONS_ENABLED_DEFAULT) === 'true';
+    if (oneClickReactionsEnabled) {
+        emojis = getOneClickReactionEmojis(state);
+    }
```

Comment:
may be this can go its own selector, this has been used more than 2, now 3 times